shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ResultSetUtil.java

```diff
@@ -224,4 +227,22 @@ private static Object convertByteArrayValue(final Object value, final Class<?> c
                 return value;
         }
     }
+
+    private static Object convertBooleanValue(final Object value) {
+        if (value instanceof Boolean) {
+            return value;
+        }
+        String stringVal = value.toString();
+        if (stringVal.length() > 0) {
+            int c = Character.toLowerCase(stringVal.charAt(0));
+            return c == 't' || c == 'y' || c == '1' || "-1".equals(stringVal);
```

Comment:
@challengeof Can we rename `c` to `firstChar`? And put constant on the left condition?